plugins/ingress_nginx_logs.yaml

```diff
@@ -0,0 +1,106 @@
+version: 0.0.1
+title: Ingress NGINX
+description: Log parser for Ingress NGINX
+parameters:
+  - name: file_path
+    type: "[]string"
+    default:
+      - "/var/log/containers/ingress-nginx-controller*.log"
+  - name: start_at
+    type: string
+    supported:
+      - beginning
+      - end
+    default: end
+  # An optional cluster name
+  - name: cluster_name
+    type: string
+    default: ""
+template: |
+  receivers:
+      filelog:
+        include:
+          {{ range $fp := .file_path }}
+          - '{{ $fp }}'
+          {{ end }}
+        start_at: {{ .start_at }}
+        resource:
+          {{ if .cluster_name }}
+          k8s.cluster.name: {{ .cluster_name }}
+          {{ end }}
+        operators:
+          # Initial log object looks like this:
+          # '{"log":"NGINX Ingress controller\n","stream":"stdout","time":"2022-05-18T01:27:30.100955955Z"}'
+
+          - type: json_parser
+
+          # Move the request message to body
+          - type: move
+            from: attributes.log
+            to: body
```

Comment:
Is there a reason we don't keep the original log line on body? It seems to break an emergent convention we have where the full, original log line is always available on the body. Wondering what your thoughts on it is.

Reply:
My thoughts are that Body is kept as the original log line in case things fail to parse, so at worst you at least have whatever you lost available on the record somewhere.

That being said; It sounds like the original record can be easily reconstructed, and we're sure of its structure, so it probably makes sense to do this in k8s.